sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/BaseSessionCatalog.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog;
+
+import org.apache.spark.annotation.Experimental;
+
+/**
+ * An interface that aggregates different catalog interfaces that can be supported
+ * by a session catalog.
+ */
+@Experimental
+public interface BaseSessionCatalog extends TableCatalog, SupportsNamespaces {
```

Comment:
I'd like to avoid a new interface.

We can do `CatalogExtension extends TableCatalog, SupportsNamespaces`, and change the method to
```
void setDelegateCatalog(CatalogPlguin delegate);
```
`CatalogManager.defaultSessionCatalog` can also be `CatalogPlugin`

Reply:
OK. I removed `BaseSessionCatalog` and modified `DelegatingCatalogExtension` to do downcasting delegate for each function.